### PR TITLE
pythontex environments added

### PR DESCRIPTION
**Pull request by ndvanforeest**
I use pythontex quite a bit, and the functionality of blacken-doc was precisely what I needed.

### CODE REVIEW

File: blacken_docs.py [hunk 1]
```diff
@@ -36,6 +36,13 @@
     r'(?P<after>^(?P=indent)\\end{minted}\s*$)',
     re.DOTALL | re.MULTILINE,
 )
+PYTHONTEX = '(pyblock|pycode|pyconsole|pyverbatim)'
+PYTHONTEX_RE = re.compile(
+    rf'(?P<before>^(?P<indent> *)\\begin{{{PYTHONTEX}}}\n)'
+    r'(?P<code>.*?)'
+    rf'(?P<after>^(?P=indent)\\end{{{PYTHONTEX}}}\s*$)',
+    re.DOTALL | re.MULTILINE,
+)
```
Comment (asottile) [inline blacken_docs.py:45]: this should probably just extend the `LATEX_RE` above instead of duplicating I would assume?

File: blacken_docs.py [hunk 2]
```diff
@@ -36,6 +36,13 @@
     r'(?P<after>^(?P=indent)\\end{minted}\s*$)',
     re.DOTALL | re.MULTILINE,
 )
+PYTHONTEX = '(pyblock|pycode|pyconsole|pyverbatim)'
+PYTHONTEX_RE = re.compile(
+    rf'(?P<before>^(?P<indent> *)\\begin{{{PYTHONTEX}}}\n)'
+    r'(?P<code>.*?)'
```
Comment (asottile) [inline blacken_docs.py:42]: would prefer the prefix here stay `rf` so that the strings line up

**Comment by ndvanforeest** [comment]
Yes, I think so, however, the minted environment takes the option "python"
whereas the pythontex environments don't. In honesty, I don't know how to
solve this in a good way, except by my proposal. However, I am really
interested to see how you tackle this point in one go.

**Comment by ndvanforeest** [comment]
On Wed, 6 May 2020 at 16:50, nicky van foreest <vanforeest@gmail.com> wrote:

> Yes, I think so, however, the minted environment takes the option "python"
> whereas the pythontex environments don't. In honesty, I don't know how to
> solve this in a good way, except by my proposal. However, I am really
> interested to see how you tackle this point in one go.
>
> On Wed, 6 May 2020 at 16:44, Anthony Sottile <notifications@github.com>
> wrote:
>
>> *@asottile* commented on this pull request.
>> ------------------------------
>>
>> In blacken_docs.py
>> <https://github.com/asottile/blacken-docs/pull/48#discussion_r420848886>:
>>
>> > @@ -36,6 +36,13 @@
>>      r'(?P<after>^(?P=indent)\\end{minted}\s*$)',
>>      re.DOTALL | re.MULTILINE,
>>  )
>> +PYTHONTEX = '(pyblock|pycode|pyconsole|pyverbatim)'
>> +PYTHONTEX_RE = re.compile(
>> +    rf'(?P<before>^(?P<indent> *)\\begin{{{PYTHONTEX}}}\n)'
>> +    r'(?P<code>.*?)'
>>
>> would prefer the prefix here stay rf so that the strings line up
>>
>
Sure, I did not think of this.

**Comment by asottile** [comment]
you can probably extend your `|`

something like

```python
PYTHONS = r'(\{python\}\{minted\}|\{(pyblock|pycode|pyconsole|pyverbatim)\})'
```

**Pull request closed without merging**